Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -0,0 +1,232 @@
+import UIKit
+@_implementationOnly import MapboxCommon_Private
+@_implementationOnly import MapboxCoreMaps_Private
+
+public enum ViewAnnotationManagerError: Error {
+    case insertionFailure(reason: String)
+    case updateFailure(reason: String)
+    case removalFailure(reason: String)
+    case optionsQueryFailure(reason: String)
```

Comment:
Not sure if we need to have an error case for the options queries.
Currently the APIs which retrieve options are returning optionals, which might be enough indication for the status. This is also thrown if we fail to unwrap the value that's coming from gl-native, but that shouldn't happen if the response was not an error.